lib/uklibid/include/uk/libinfo.h

```diff
@@ -0,0 +1,102 @@
+#ifndef __LIBUKLIBID_INFO_H__
+#define __LIBUKLIBID_INFO_H__
+
+#include <uk/arch/types.h>
+#include <uk/essentials.h>
+
+/*
+ * Library information is stored in the special section " .uk_libinfo". For
+ * each library there is one `struct uk_libid_info_hdr` generated. Each header
+ * can contain multiple `struct uk_libid_info_rec` records. A library name
+ * record (`UKLI_REC_LIBNAME`) is the only mandatory record type for each
+ * library.
+ * Additonally, a single header is created for global information, which differs
+ * from the library headers in that it does not contain a record for a library
+ * name.
+ *
+ * Layout for each information header:
+ *
+ *                              _____________
+ *  +--------------------------+             \
+ *  | struct uk_libid_info_hdr |              |
+ *  |                          |               > hdr->len
+ *  +--------------------------+              |
+ *  | struct uk_libid_info_rec |              |
+ *  +--------------------------+ -.           |
+ *  | struct uk_libid_info_rec |   > rec->len |
+ *  |                          |  |           |
+ *  +--------------------------+ -`           |
+ *  :          . . .           :              :
+ *  +--------------------------+              |
+ *  | struct uk_libid_info_rec |              |
+ *  |                          |              |
+ *  +--------------------------+ ____________/
+ */
+
+/*
+ * NOTE: The layout version needs only be updated if the structs
+ *       `uk_libid_info_rec` and `uk_libid_info_hdr` are updated.
+ */
+ #define UKLI_LAYOUT              0x0001
+
+/*
+ * NOTE: In order to keep backwards compatibility with pre-compiled
+ *       libraries, this list should only be extended with new entries.
+ */
+#define UKLI_REC_LIBNAME          0x0001 /* C-string, absent for global hdr */
+#define UKLI_REC_COMMENT          0x0002 /* C-string */
+#define UKLI_REC_VERSION          0x0003 /* C-string */
+#define UKLI_REC_LICENSE          0x0004 /* C-string */
+#define UKLI_REC_GITDESC          0x0005 /* C-string */
```

Comment:
The idea was to provide the framework with this PR and extend it with functionality subsequently. The PR misses currently a feature to execute a script or command for each library to retrieve its Git version. Similarly are the licenses. Do we want to detect SPDX headers or should we use a defined place per library to look this information up? My idea was that these fields could be manually already be used with custom code and would have a well-defined place to store. Further implementation for automating this could be done afterwards.

Reply:
The git part is much easier than the licensing which can be quite tricky for libraries that involve multiple licenses, and may require generating a license report using automated tooling like FOSSology or BlackDuck.

But okay, if the intention was to have it as a placeholder, then anyone who needs it will open a PR 👍🏼

Reply:
👍🏻